pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,8 +102,10 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
+		e.recorder.Eventf(config, kapi.EventTypeWarning, "Failed", "The %s-hook failed (%s), aborting deployment %s/%s", suffix, label, err, deployment.Namespace, deployment.Name)
 		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
 	case deployapi.LifecycleHookFailurePolicyIgnore:
+		e.recorder.Eventf(config, kapi.EventTypeWarning, "Failed", "The %s-hook failed but ignored, deployment %s/%s will continue", suffix, label, err, deployment.Namespace, deployment.Name)
```

Comment:
Same comment as above. We probably don't need the `warning` prefix for the event message.